lib/streamlit/runtime/caching/cache_utils.py

```diff
@@ -499,6 +531,15 @@ def _make_function_key(cache_type: CacheType, func: FunctionType) -> str:
         source_code, hasher=func_hasher, cache_type=cache_type, hash_source=func
     )
 
+    # Include relevant closure values in the hash
+    closure_values = _get_relevant_closure_values(func)
```

Comment:
Creating function hash like this doesn't feel right to me. My thought is like:

```py
captured_var = 1

@st.cache_data()
def foo():
  return captured_var + 1

assert foo() == 2
captured_var = 2
assert foo() == 3   # can captured_var's value at the time of st.cache() call help  get this right?
```

I'm guessing it would be difficult if ever possible to automatically get closure values right. We can't assume all captured values are hashable, their immutability or referential transparency etc.

Reply:
Yea. Things get noticably more complicated and delicate to change. I am going to close this issue and see if the issue gets a lot more upvotes.